packages/client/lib/rpc/modules/engine.ts

```diff
@@ -989,4 +1014,107 @@ export class Engine {
       blobs: bundle.blobs.map((blob) => '0x' + blob.toString('hex')),
     }
   }
+
+  /**
+   * Returns a list of engine API endpoints supported by the client
+   */
+  private getCapabilities(_params: []): string[] {
+    const caps = Object.getOwnPropertyNames(Engine.prototype)
+    const engineMethods = caps.filter((el) => el !== 'constructor' && el !== 'getCapabilities')
+    return engineMethods.map((el) => 'engine_' + el)
+  }
+
+  /**
+   *
+   * @param params a list of block hashes as hex prefixed strings
+   * @returns an array of ExecutionPayloadBodyV1 objects or null if a given execution payload isn't stored locally
+   */
+  private async getPayloadBodiesByHashV1(
+    params: [[Bytes32]]
+  ): Promise<(ExecutionPayloadBodyV1 | null)[]> {
+    if (params[0].length > 32) {
+      throw {
+        code: TOO_LARGE_REQUEST,
+        message: 'More than 32 execution payload bodies requested',
+      }
+    }
+    const hashes = params[0].map((hash) => toBuffer(hash))
+    const blocks: (ExecutionPayloadBodyV1 | null)[] = []
+    for (const hash of hashes) {
+      try {
+        const block = await this.chain.getBlock(hash)
+        const transactions: string[] = []
+        for (const txn of block.transactions) {
+          transactions.push('0x' + txn.serialize().toString('hex'))
+        }
+        let withdrawals
+        if (block._common.gteHardfork(Hardfork.Shanghai)) {
+          withdrawals = []
+          for (const withdrawal of block.withdrawals!) {
+            withdrawals.push(withdrawal.toJSON() as WithdrawalV1)
+          }
+        } else {
+          withdrawals = null
+        }
+        blocks.push({ transactions, withdrawals })
+      } catch {
+        blocks.push(null)
+      }
+    }
+    return blocks
+  }
+
+  /**
+   *
+   * @param params an array of 2 parameters
+   *    1.  start: Bytes8 - the first block in the range
+   *    2.  count: Bytes8 - the number of blocks requested
+   * @returns an array of ExecutionPayloadBodyV1 objects or null if a given execution payload isn't stored locally
+   */
+  private async getPayloadBodiesByRangeV1(
+    params: [Bytes8, Bytes8]
+  ): Promise<(ExecutionPayloadBodyV1 | null)[]> {
+    const start = BigInt(params[0])
+    let count = BigInt(params[1])
+    if (count > BigInt(32)) {
+      throw {
+        code: TOO_LARGE_REQUEST,
+        message: 'More than 32 execution payload bodies requested',
+      }
+    }
+
+    if (count < BigInt(1) || start < BigInt(1)) {
+      throw {
+        code: INVALID_PARAMS,
+        message: 'Start and Count parameters cannot be less than 1',
+      }
+    }
+    const currentChainHeight = (await this.chain.getCanonicalHeadHeader()).number
```

Comment:
we should be able to use chain.latest.height or something since its build similarly, could avoid wait/load from db